kahuna/public/js/image/streamService.js

```diff
@@ -0,0 +1,48 @@
+import Rx from 'rx';
+
+function imageStream(images_) {
+
+    const images = new Set();
```

Comment:
It's easier to add  / delete form a set than from an array... this is more of an internal representation, hiding it from clients of this as they shouldn't have to worry how things are added / removed.

We then return an array as that where you get to do the nice functional stuff to it.